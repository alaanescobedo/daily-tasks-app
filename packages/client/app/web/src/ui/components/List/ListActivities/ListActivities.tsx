import styles from './ListActivities.module.css'
import { ActivityCard } from '@components/Card/Activity'
import { Scroll } from '@components/Scroll'
import { TaskI } from '@modules/task/task.interface'

export interface ListActivitiesProps {
  tasks: TaskI[]
  day: string
}

export const ListActivities = ({ tasks, day }: ListActivitiesProps): JSX.Element => {
  return (
    <Scroll>
      <ul className={styles.container}>
        {tasks.map(task => (
          <ActivityCard title={task.title} key={task.entityId} />
        ))}
      </ul>
    </Scroll>
  )
}
