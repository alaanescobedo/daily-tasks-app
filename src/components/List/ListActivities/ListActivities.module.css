.container {
    padding: 20px;
}

.list_item {
    color: rgb(var(--text));
    padding-left: 1.2rem;
    flex: 1;
}