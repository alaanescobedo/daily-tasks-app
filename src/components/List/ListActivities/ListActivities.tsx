import { Link } from 'react-router-dom'
import { Task } from '@hooks'
import styles from './ListActivities.module.css'

export interface ListActivitiesProps {
  tasks: Task[]
  day: string
}

export const ListActivities = ({ tasks, day }: ListActivitiesProps): JSX.Element => {
  return (
    <ul className={styles.container}>
      {tasks.map(task => (
        <span style={{ display: 'flex', gap: '1rem' }} key={task.entityId}>
          <li className={styles.list_item}>{task.title}</li>
          <span style={{ cursor: 'pointer' }}>Remove</span>
          <Link to={`/activities/update/${day}`}>Update</Link>
        </span>
      ))}
    </ul>
  )
}
