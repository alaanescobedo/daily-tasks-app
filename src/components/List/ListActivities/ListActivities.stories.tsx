import { ComponentStory, ComponentMeta } from '@storybook/react'

import { ListActivities as ListActivitiesComponent, ListActivitiesProps } from './ListActivities'

const ListActivitiesStory: ComponentMeta<typeof ListActivitiesComponent> = {
  title: 'List/List Of Activities',
  component: ListActivitiesComponent
}
export default ListActivitiesStory

const Template: ComponentStory<typeof ListActivitiesComponent> = (args) => <ListActivitiesComponent {...args} />

export const ListOfActivities = Template.bind({})
const ListActivitiesArgs: ListActivitiesProps = {
  day: '2022-04-11',
  tasks: [
    {
      title: 'Task 1',
      createdAt: '2020-01-01',
      entityId: Math.random().toString(),
      scheduledFor: '2022-04-11',
      status: 'Pending',
      userID: 'user123'
    },
    {
      title: 'Task 2',
      createdAt: '2020-01-01',
      entityId: Math.random().toString(),
      scheduledFor: '2022-04-11',
      status: 'Pending',
      userID: 'user123'
    }
  ]
}
ListOfActivities.args = ListActivitiesArgs
