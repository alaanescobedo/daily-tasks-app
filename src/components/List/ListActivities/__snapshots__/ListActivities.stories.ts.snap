// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots List/List Of Activities List Of Activities 1`] = `
Array [
  " ",
  <div
    style={
      Object {
        "--background": "2, 40, 60",
        "--highlight": "255,  202,  16",
        "--opacity-1": 0.1,
        "--opacity-7": 0.7,
        "--primary": "206, 180, 249",
        "--secondary": "189, 231, 255",
        "--tertiary": "176, 239, 165",
        "--text": "245, 245, 245",
        "0.1": "--opacity-1",
        "0.7": "--opacity-7",
      }
    }
  >
     
    <ul
      className="container"
    >
      <span
        style={
          Object {
            "display": "flex",
            "gap": "1rem",
          }
        }
      >
        <li
          className="list_item"
        >
          Task 1
        </li>
        <span
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          Remove
        </span>
        <a
          href="/activities/update/2022-04-11"
          onClick={[Function]}
        >
          Update
        </a>
      </span>
    </ul>
     
  </div>,
  " ",
]
`;
